src/Concerns/JsonSerializer.php

```diff
@@ -25,14 +25,17 @@ protected function setValues($values)
     {
```

Comment:
@floranpagliai As part of this, I think we should throw an error here if we get a value that is not valid for `foreach`.

Reply:
The only non-valid value was `null`, and this was because of the recursion
```
if (isset($this->keys[$key])) {
    $className = $this->keys[$key];
    if (method_exists($className, 'build')) {
        $this->{ $key } = $className::build($value);
    } else {
        $this->{ $key} = new $className($value);
    }
}
```
Because `$className::build(null);` was made

This is not possible anymore with 
```
if (null === $value) {
    continue;
}
```

Do you want to also add a check
```
protected function setValues($values)
{
    if (null === $values) {
        throw new \Exception();
    }
```
? This could be considered as a BC break because previously
```
setValues(null);
```
is working for some entities.

So I would recommend to not introduce such change in a bugfix.

Reply:
> This could be considered as a BC break because previously
setValues(null);

> is working for some entities.

I'm not sure if it was working. Can you identify the entities for which it was functioning correctly?  Because as far as I can understand (which may not be a lot - so feel free to correct me) `setValues` will do nothing if `$values` is anything other than an array or object

Reply:
>  Can you identify the entities for which it was functioning correctly? Because as far as I can understand (which may not be a lot - so feel free to correct me) setValues will do nothing if $values is anything other than an array or object

I will rephrase: Currently the code, with setValues(null), is not always throwing an error in production.

For instance, if you don't use the method isHigherPriority
https://github.com/Flagsmith/flagsmith-php-client/blob/f044353b997fca7e4a1f13a67e1422cc577ea8b3/src/Engine/Features/FeatureStateModel.php#L248
You will not have an exception, even if setValues(null) was called.
But if you use isHigherPriority you get an exception (which we got).

This PR solve our issue. I cannot ensure you it will solve 100% of other wrong values (even if I believe it).

Merging this PR like this will:
- Solve our issue
- Untouch other issue if they exist

Adding the exception to our PR will
- Solve our issue
- Throw new exception (but maybe not) if other issues exists even if people were not encountering them.

I think throwing an exception for bad values is a good idea @gagantrivedi but I can do it in another PR. My recommendation would be to
- Merge this
- Release a patch version (people will be sure to have no bad impact on bumping the version)
- I add the exception
- Release a major version (or minor depending on the point of view)

Reply:
Yeah, that makes sense